backend/src/Good/good.models.ts

```diff
@@ -35,8 +36,8 @@ class Good {
      * Retrieve a good from an id
      * @param id the id of the good
      */
-    public static async findById(id: number): Promise<AnyGood> {
-        const existing = await db
+    public static async findById(id: number) {
+        let schema: AnyGood | null = await db
```

Comment:
```suggestion
        let schema: AnyGood | undefined = await db
```
The db returns undefined if nothing was returned